File: platform/College-1/Litcoder/Week 2/Code/lab7.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        
        int target = in.nextInt();
        in.nextLine(); // Consume the newline character
        
        String[] num = in.nextLine().split(" ");
        int size = num.length;
        int count = 0;

        // Possible: NumberFormatException
        try {
            for (int ptr1 = 0; ptr1 < size; ptr1++) {
                int num1 = Integer.parseInt(num[ptr1]);
                for (int ptr2 = ptr1 + 1; ptr2 < size; ptr2++) {
                    int num2 = Integer.parseInt(num[ptr2]);
                    if ((num1 + num2) % target == 0) {
                        count++;
                    }
                }
            }
        } catch (Exception e) {
            System.exit(1); // Exit the program if an exception is caught
        }

        System.out.print(count);
        in.close();
    }
}

File: platform/College-1/Litcoder/Week 2/Code/lab7.py
def findSumPairs(arr, divisor):
    count = 0
    for ptr1 in range(len(arr)):
        for ptr2 in range(ptr1 + 1, len(arr)):
            if (arr[ptr1] + arr[ptr2]) % divisor == 0:
                count += 1
    return count

# Read the divisor
divisor = int(input())

# Read the array of integers
arr = list(map(int, input().split()))

# Print the result
print(findSumPairs(arr, divisor), end='')
